### PR TITLE
Fix timeout, add Closable::onClose()

### PR DESCRIPTION
**Pull request by bileslav**

### CODE REVIEW

File: src/Internal/PsrInputStream.php [hunk 1]
```diff
@@ -29,6 +29,11 @@ public function __construct(StreamInterface $stream, int $chunkSize = self::DEFA
         $this->chunkSize = $chunkSize;
     }
 
+    public function onClose(\Closure $onClose): never
+    {
+        throw new \Error("Not implemented");
```
Comment (kelunik) [inline src/Internal/PsrInputStream.php:34]: This can just call `$this->stream->onClose($onClose);`

**Comment by kelunik** [comment]
Thanks! You don't need to close this PR and open another one, you can just edit your branch further. We can always squash the commits at the end.

**Comment by bileslav** [comment]
> We can always squash the commits at the end.

Okay, thanks for the tip!

File: src/Internal/PsrInputStream.php (hunk 1 above)
Comment (kelunik) [reply src/Internal/PsrInputStream.php:34]: Nevermind.

**Comment by kelunik** [comment]
Thanks!

**Pull request merged**